vespajlib/src/main/java/com/yahoo/slime/ArrayValue.java

```diff
@@ -55,15 +66,22 @@ public void prepareFor(ArrayValue self, Type type) {
                 self.impl = new GenericImpl(this, self.used);
             }
         }
+        private static double [] grow(double [] arr) {
+            double [] v = new double[arr.length << 1];
+            System.arraycopy(arr, 0, v, 0, arr.length);
+            return v;
+        }
         public Value add(Value value, int used) {
+            return add(value.asDouble(), used);
```

Comment:
this could also return NixValue.invalid as it will never be called

Reply:
Fixed